### DISCUSSION TITLE
Any reason why Polly create histogram metrics with units as milliseconds and type as double?

### DISCUSSION THREAD

**Discussion by lvxiaoxin** [App-vNext/Polly#2363] [Q&A] [answered]
Hi Polly expert:

I noticed that Polly is creating `resilience.polly.strategy.attempt.duration` and `resilience.polly.pipeline.duration` with units as `ms`, histogram type as `double`. Any reason for that? Per my understanding, usually we would use `double` when unit is seconds, or use `long` as type when unit is milliseconds.

**Comment by martincostello** [comment, marked as answer]
The type is the same as the type of the value we report:

https://github.com/App-vNext/Polly/blob/12fe5690ec3e7a6365eb8c9e4148342f596315bc/src/Polly.Extensions/Telemetry/TelemetryListenerImpl.cs#L143

`Duration` is a `TimeSpan`:

https://github.com/App-vNext/Polly/blob/12fe5690ec3e7a6365eb8c9e4148342f596315bc/src/Polly.Core/Telemetry/ExecutionAttemptArguments.cs#L31-L34

[`TimeSpan.TotalMilliseconds`](https://learn.microsoft.com/dotnet/api/system.timespan.totalmilliseconds) is a `double`.
